gtest_parallel.py

```diff
@@ -204,14 +204,39 @@ def _logname(output_dir, test_binary, test_name, execution_number):
     # Store logs to temporary files if there is no output_dir.
     if output_dir is None:
       (log_handle, log_name) = tempfile.mkstemp(prefix='gtest_parallel_',
-                                                suffix=".log")
+                                                suffix='.log')
       os.close(log_handle)
       return log_name
 
-    log_name = '%s-%s-%d.log' % (Task._normalize(os.path.basename(test_binary)),
-                                 Task._normalize(test_name), execution_number)
+    # The log file name is a combination of a test binary name and a test name.
+    # If the log file name exceeds OS limitations it will be truncated.
+    MKSTEMP_RANDOM_LEN = 12  # length of mkstemp random string being added
+    MAX_PREFIX_LENGTH = 240  # Limit for file name mkstemp prefix
+
+    prefix = '%s-%s' % (Task._normalize(os.path.basename(test_binary)),
+                        Task._normalize(test_name))
+    suffix = '-%d.log' % execution_number
+    log_name = os.path.join(output_dir, prefix + suffix)
+    try:
+      # check if log can be created or exists already
+      if not os.path.exists(log_name):
+        os.makedirs(output_dir, exist_ok=True)
+        with open(log_name, 'w+'):
+          pass
+        os.remove(log_name)
+    except OSError as os_error:
+      # truncate file name if error is Errno 36: File name too long
+      if os_error.errno == 36:
+        prefix_length = ((os.statvfs('.').f_namemax -
+                          MKSTEMP_RANDOM_LEN) if hasattr(os, 'statvfs') else MAX_PREFIX_LENGTH) - len(suffix)
```

Comment:
I think this would read easier as: if hasattr(..): prefix_length = ..., i.e. avoid the ternary. as it's long.